### PR TITLE
Use test vectors provided by Fer, fix bugz

### PR DESCRIPTION
**Pull request by gimer**

### CODE REVIEW

**Comment by fboucquez** [comment]
Nice one, Travis is running test tests.

https://travis-ci.com/github/nemtech/catbuffer-generators/jobs/482384310#L3987

Eventually, a py sdk would improve the testing by validating each builder and their fields. 

An improvement to the vector test in the future is adding fields, so once a hex is parsed, assert that a field in the created builder is the same as an expected one from the file.

**Review by fboucquez — APPROVED** [review]

File: generators/python/test_TransferTransactionBuilder.py [hunk 1]
```diff
@@ -88,17 +88,17 @@ def test_serialize(self):
...
     else:
-        print("DETAIL: Missing module path '{0}'.".format(builderModulePath))
-        print("POSSIBLE CAUSES:")
-        print("    1. The '{0}' module is not in the search path for module files.".format(builderModulePath))
-        print("       Try: Mark Directory catbuffer as Sources Root or check PYTHONPATH environment variable.")
+        print('DETAIL: Missing module path \'{0}\'.'.format(builderModulePath))
+        print('POSSIBLE CAUSES:')
+        print('    1. The \'{0}\' module is not in the search path for module files.'.format(builderModulePath))
+        print('       Try: Mark Directory catbuffer as Sources Root or check PYTHONPATH environment variable.')
     raise
```
Comment (gimer) [inline generators/python/test_TransferTransactionBuilder.py:104]: @fboucquez @Jaguar0625 not sure if I should leave it as is or rather remove completely

Comment (Jaguar0625) [reply generators/python/test_TransferTransactionBuilder.py:104]: would remove

File: generators/python/test_VectorTest.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+
+def prepare_test_cases():
+    cases = []
+    for f in Path('vector').iterdir():
+        cases += read_test_vectors_file(f)
+    return cases
+
+
+def to_hex_string(binary: bytes):
```
Comment (Jaguar0625) [inline generators/python/test_VectorTest.py:21]: what is `binary: bytes` syntax?

File: generators/python/test_VectorTest.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+@pytest.mark.parametrize('item', prepare_test_cases(), ids=generate_pretty_id)
+def test_serialize(item):
+    builderName = item['builder']
+    comment = item['comment'] if 'comment' in item else ''
+    payload = item['payload']
+
+    builderModule = importlib.import_module('symbol_catbuffer.{}'.format(builderName))
+    builderClass = getattr(builderModule, builderName)
+    print(item['payload'])
```
Comment (Jaguar0625) [inline generators/python/test_VectorTest.py:48]: intended?

**Review by Jaguar0625 — APPROVED** [review]

File: generators/python/test_VectorTest.py (hunk 2 above)
Comment (gimer) [reply generators/python/test_VectorTest.py:21]: type hints, they are present in generated code, so I used them here as well: https://docs.python.org/3/library/typing.html

**Comment by gimer** [comment]
thx for reviewing

**Pull request merged**